### DISCUSSION TITLE
Create Observations over MQTT (subscription)

### DISCUSSION THREAD

**Discussion by mhaertel** [FraunhoferIOSB/FROST-Server#1647] [Q&A] [answered]
I am new to OGC SensorThings API and MQTT. I studied the OGC documents but I am puzzled about the result I get. I can configure some sensors to send updates through MQTT and I always receive the same message in JSON format. 

This is taken from the `/var/log/tomcat9/SensorThingsService.log`: `ERROR d.f.i.i.f.mqtt.MqttManager - Creating entity via MQTT failed`

The JSON I receive is as follows:
`

```
{
	"time":"2023-07-18T18:49:48.604Z",	
	"device":{
		"id":"sensecap-guitar-moonlight-coffee",
		"name":"SenseCAP-guitar-moonlight-coffee",
		"driver":"",
		"labels":[
			{
				"id":1,
				"value":
				"Temperaturmessung",
				"hide_nav":false
			}
		],
		"state":{
			"desired":{},
			"reported":{}
		},
		"auto_activated":false,
		"custom_fields":{},
		"created_at":"2023-07-04T08:18:02.103416Z",
		"device_updated_at":"2023-07-11T13:59:38.265684Z",
		"state_changed_at":"2023-07-18T18:44:50.905659Z",
		"last_data_received":{
			"received_at":"2023-07-18T18:49:48.883155789Z",
			"raw":"0101100e6f00000102106e960000953c"
		},
		"connectivity":{
			"connections":[
				{"network_server_id":1,
				"network_device_id":"2CF7F1C04230004E",
				"network_server":{
					"id":1,
					"name":"NetCologne Network"}
				}
			]
		}
	},
	"network_server":{
		"Device":"2CF7F1C04230004E",
		"Connection":"1"
	},
	"state":{},
	"errors":null,
	"version":"0.0"
}
```

`

From what I found in the internet it is not really fitting to Observations only and the message that comes together with the JSON is: 

`{"code":400,"type":"error","message":"Unknown field: time\n at [Source: (StringReader); line: 1, column: 10]"}`

Is there anything I can do to make FROST Server understand that JSON? Is the "ThingsHub" expected to send that type of JSON as Observations?

**Comment by hylkevds** [comment, marked as answer]
FROST implements the OGC SensorThings API, so you have to send it messages that conform to that standard.
If your sensor can't send messages according to the standard you'll have to write a converter.

**Reply by mhaertel** [reply]
Thank you very much. That confirms that the sender of the data doesn't conform to the Standards.
